### ISSUE TITLE
Preparing Custom Dataset

### ISSUE THREAD

**Issue by anmspro** [matterport/Mask_RCNN#2259]
I have a dataset folder which contains three folders named images, tags, masks. Each of these folders contains images. For example:
In images:
<img src="https://user-images.githubusercontent.com/33668152/85875652-5c80bc80-b7f6-11ea-9ae1-4981262e5911.jpg"  width="250" />
In tags:
<img src="https://user-images.githubusercontent.com/33668152/85875685-699dab80-b7f6-11ea-8ce6-7145ed1cf9e0.png"  width="250" />
In masks:
<img src="https://user-images.githubusercontent.com/33668152/85875714-77533100-b7f6-11ea-94ce-310c862bdddd.png"  width="250" />

So, I have the bounding box region cropped and a mask from that cropped region. All these are in image format.

How do I prepare this dataset to use for training on a pretrained Mask-RCNN model?

**Comment by konstantin-frolov** [comment]
See information about pycococreator.

**Comment by anmspro** [comment]
> See information about pycococreator.

Can you please elaborate?

**Comment by konstantin-frolov** [comment]
> Can you please elaborate?

[Create your own COCO-style datasets](https://patrickwasp.com/create-your-own-coco-style-dataset/)

**Comment by anmspro** [comment]
> > Can you please elaborate?
> 
> [Create your own COCO-style datasets](https://patrickwasp.com/create-your-own-coco-style-dataset/)

Hey, man!
I have read the blog. I tried to use the code with some modification. Still couldn't generate dataset.
I got an error while trying it. Could you please take a look and provide some guidelines? 
Here's the link - https://github.com/waspinator/pycococreator/issues/34

**Comment by gethubwy** [comment]
> See information about pycococreator.

Hello, Excuse me, I want to know how to train a new mode, I use the software named "labelme" to label my data, I want to train a new model starting from pre-trained COCO weights, but I don't know how to write the **annotation file in validate set**, are there tools to help me? Thank you.

**Comment by konstantin-frolov** [comment]
> > See information about pycococreator.
> 
> Hello, Excuse me, I want to know how to train a new mode, I use the software named "labelme" to label my data, I want to train a new model starting from pre-trained COCO weights, but I don't know how to write the **annotation file in validate set**, are there tools to help me? Thank you.

Validation annotation is the same as train

**Comment by gethubwy** [comment]
> > > See information about pycococreator.
> > 
> > 
> > Hello, Excuse me, I want to know how to train a new mode, I use the software named "labelme" to label my data, I want to train a new model starting from pre-trained COCO weights, but I don't know how to write the **annotation file in validate set**, are there tools to help me? Thank you.
> 
> Validation annotation is the same as train

actually，they are different! for example:
instances_train2017.json:
{   "license": 1,
    "file_name": "000000536292.jpg",
    "coco_url": "http://images.cocodataset.org/train2017/000000536292.jpg
    "height": 425,
    "width": 640,
    "date_captured": "2013-11-18 13:14:31
    flickr_url": "http://farm4.staticflickr.com/3689/9154491588_866ac8791e_z.jpg",
    "id": 536292
}


well the  instances_val2017.json is: 
{"segmentation": [
    [314.85,365.65,305.09,379.74,295.88,378.12,293.17,393.29,357.11,390.58,391.24,387.87,382.03,365.65],
    [289.38,375.95,266.63,370.53,262.29,388.95,271.5,393.29,280.17,392.2,282.34,392.2],
[317.01,341.27,308.89,330.44,306.18,320.68,307.8,312.56,311.6,288.72,332.73,283.3,350.06,289.26,350.06,300.1,352.77,305.51,351.15,327.73,349.52,339.11,351.15,342.9]],
    "area": 4868.703549999998,
    "iscrowd": 0,
    "image_id": 545594,
    "bbox": [262.29,283.3,128.95,109.99],
    "category_id": 1,
    "id": 207363
}

**Comment by konstantin-frolov** [comment]
In your example in validatiton annotations added only area field. 
I use identical annotation for train and Val. It’s work fine.
